src/main/python/rlbot/setup_manager.py

```diff
@@ -120,14 +125,38 @@ def run(self):
                 break
             try:
                 single_agent_metadata = self.agent_metadata_queue.get(timeout=1)
-                self.agent_metadata_map[single_agent_metadata['index']] = single_agent_metadata
+                self.start_or_update_helper_process(single_agent_metadata)
+                self.agent_metadata_map[single_agent_metadata.index] = single_agent_metadata
                 configure_processes(self.agent_metadata_map, self.logger)
             except queue.Empty:
                 pass
             except Exception as ex:
                 self.logger.error(ex)
                 pass
 
+    def start_or_update_helper_process(self, agent_metadata: AgentMetadata):
+        """
+        Examines the agent metadata to see if the agent needs a helper process. If the process is not running yet,
+        create the process. Once the process is running, feed the agent metadata to it.
+
+        If a process is created here, the pid will be added to the agent metadata.
+        """
+
+        helper_req = agent_metadata.helper_process_request
+
+        if helper_req is not None:
+            if helper_req.key not in self.helper_process_map:
+                metadata_queue = mp.Queue()
```

Comment:
I'd consider moving the `metadata_queue` creation into a separate method, e.g. like

```python
if helper_req.key not in self.helper_process_map:
    self.helper_process_map[helper_req.key] = SetupManager.create_metadata_queue(agent_metadata, self.quit_event)
```

And with that, I'd also consider moving `run_helper_process` and `create_metadata_queue` into a separate file, for example `helper_utils.py`

Reply:
Refactored into a separate file.